shared/api/instance.go

```diff
@@ -374,3 +378,80 @@ type InstanceSource struct {
 	// API extension: instance_allow_inconsistent_copy
 	AllowInconsistent bool `json:"allow_inconsistent" yaml:"allow_inconsistent"`
 }
+
+// InstanceEFI represents the EFI vars of a LXD virtual machine instance.
+//
+// swagger:model
+//
+// API extension: instances_efi_vars.
+type InstanceEFI struct {
+	// EFI variables map
+	// Hashmap key format is <efi-variable-name>-<UUID>
+	// Example: AmdAcpiVar-79941ecd-ed36-49d0-8124-e4c31ac75cd4
+	Variables map[string]*InstanceEFIVariable `json:"variables" yaml:"variables"`
+}
+
+// InstanceEFIVariable represents a EFI variable entry
+//
+// swagger:model
+//
+// API extension: instances_efi_vars.
+type InstanceEFIVariable struct {
+	// EFI variable data (HEX-encoded)
+	Data string `json:"data" yaml:"data"`
+
+	// EFI variable attributes
+	Attr uint32 `json:"attr" yaml:"attr"`
+
+	// EFI variable timestamp (HEX-encoded)
+	Timestamp string `json:"timestamp" yaml:"timestamp"`
+
+	// EFI variable digest (HEX-encoded)
+	Digest string `json:"digest" yaml:"digest"`
+}
+
+// CheckValid checks whether the InstanceEFI structure is valid.
+//
+// API extension: instances_efi_vars.
+func (e InstanceEFI) CheckValid() error {
```

Comment:
I quite like this idea of adding validation on the API struct itself. 
Just a nit, but in many places where we've done validation we have it `Validate() error` so perhaps we could do this here to stick with the convention?